src/bigunit.ts

```diff
@@ -286,15 +286,33 @@ export class BigUnit {
    * @returns BigUnit with the maximum value
    */
   public static max(unit1: BigUnit, unit2: BigUnit): BigUnit {
-    return unit1.gt(unit2) ? unit1 : unit2;
+    return this.compareUnits(unit1, unit2, (u1, u2) => u1.gt(u2));
   }
 
   /**
    * @description Return the minimum of the two values
    * @returns BigUnit with the minimum value
    */
   public static min(unit1: BigUnit, unit2: BigUnit): BigUnit {
-    return unit1.lt(unit2) ? unit1 : unit2;
+    return this.compareUnits(unit1, unit2, (u1, u2) => u1.lt(u2));
+  }
+
+  /**
+   * @description Compares two BigUnit objects based on a comparison function (min or max)
+   * @returns BigUnit based on the comparison function
+   */
+  private static compareUnits(
+    unit1: BigUnit,
+    unit2: BigUnit,
+    comparisonFunc: (u1: BigUnit, u2: BigUnit) => boolean,
+  ): BigUnit {
+    if (unit1.eq(unit2)) {
+      if (unit1.precision === unit2.precision) {
+        return unit1.name.localeCompare(unit2.name) <= 0 ? unit1 : unit2;
```

Comment:
I think this name comparison should not be imposed. The `comparisonFunc` should take care of such cases